### PR TITLE
Primary caching 5: 2D & 3D point clouds

### PR DESCRIPTION
**Pull request by teh-cmc**
Integrates the cached APIs with the 2D & 3D spatial views, which is a pretty tough thing to do because there's a lot of abstraction going on in there.

`main` vs. cache disabled vs. cache enable (5950X, Arch):
```
group                        main                                    primcache_5_uncached                        primcache_5_cached                  
-----                        ----                                    --------------------                        ------------------                  
Points3D/load_all            1.68     10.1±0.14ms  94.2 MElem/sec        1.00      6.0±0.07ms 157.9 MElem/sec    1.01      6.1±0.06ms 155.7 MElem/sec
Points3D/load_colors         1.44      3.8±0.02ms 252.6 MElem/sec        1.00      2.6±0.05ms 364.0 MElem/sec    1.07      2.8±0.06ms 339.5 MElem/sec
Points3D/load_picking_ids    15.16  1859.6±7.01µs 512.9 MElem/sec        1.01    124.3±3.92µs   7.5 GElem/sec    1.00    122.7±3.86µs   7.6 GElem/sec 
Points3D/load_positions      2.29    420.1±0.76µs   2.2 GElem/sec        1.03    189.3±7.44µs   4.9 GElem/sec    1.00    183.4±5.56µs   5.1 GElem/sec 
Points3D/load_radii          1.46      3.3±0.04ms 290.1 MElem/sec        1.05      2.4±0.03ms 404.8 MElem/sec    1.00      2.2±0.00ms 423.9 MElem/sec
Points3D/query_archetype     2.51    676.1±7.59ns        ? ?/sec     15859.98      4.3±0.06ms         ? ?/sec    1.00    268.9±3.39ns         ? ?/sec 
```

---

Part of the primary caching series of PR (index search, joins, deserialization):
- #4592
- #4593
- #4659
- #4680 
- #4681
- #4698
- #4711
- #4712
- #4721 
- #4726 

---

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4592/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4592/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4592/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4592)
- [Docs preview](https://rerun.io/preview/5cfd5d661b5afba97cd24a6e87840458c6bfbc86/docs) 
- [Examples preview](https://rerun.io/preview/5cfd5d661b5afba97cd24a6e87840458c6bfbc86/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_space_view_spatial/benches/bench_points.rs [hunk 1]
```diff
@@ -51,88 +51,135 @@ fn bench_points(c: &mut criterion::Criterion) {
...
+                >(
+                    true,
+                    &store,
+                    &latest_at,
+                    &ent_path,
+                    |(_, keys, _, _, _, _, _, _)| {
+                        assert_eq!(keys.as_slice().len(), NUM_POINTS);
+                    },
+                )
+                .unwrap();
```
Comment (Wumpf) [inline crates/re_space_view_spatial/benches/bench_points.rs:80]: a bit hard to read: What is `pov1` and what does the `true` parameter do

more local: strange selection of types that isn't under a `use`

Comment (teh-cmc) [reply crates/re_space_view_spatial/benches/bench_points.rs:80]: I've added a comment for the `true` param and imported `KeypointId` & `ClassId`.

I can't really think of any way  to make the `povN_compM` clearer that what the docstring of the method already says :/

File: crates/re_space_view_spatial/benches/bench_points.rs
Comment (teh-cmc) [inline crates/re_space_view_spatial/benches/bench_points.rs:1]: This needs to run for cached & uncached + we need a comparison report for `main` vs. `cached` vs `uncached`.

File: crates/re_space_view_spatial/src/visualizers/points3d.rs [hunk 2]
```diff
@@ -231,85 +257,136 @@ pub struct LoadedPoints {
...
+    ) -> Vec<PickingLayerInstanceId> {
         re_tracing::profile_function!();
-        let iterator = arch_view
-            .iter_instance_keys()
-            .map(picking_id_from_instance_key);
+        bytemuck::cast_slice(instance_keys).to_vec()
+    }
+
+    /// Resolves all annotations and keypoints for the given entity view.
+    fn process_annotations_and_keypoints(
```
Comment (Wumpf) [inline crates/re_space_view_spatial/src/visualizers/points3d.rs:345]: previously we could share this code between visualizers. Why forfeit that by passing the component data as a whole? It's a fairly large chunk, so definitely worth it I'd say

File: crates/re_space_view_spatial/src/visualizers/entity_iterator.rs [hunk 3]
```diff
@@ -110,3 +109,125 @@ where
...
     Ok(())
 }
+
+// ---
+
+use re_query_cache::external::{paste::paste, seq_macro::seq};
+
+macro_rules! impl_process_archetype {
+    (for N=$N:expr, M=$M:expr => povs=[$($pov:ident)+] comps=[$($comp:ident)*]) => { paste! {
+        #[doc = "Cached implementation of [`process_archetype_views] for `" $N "` point-of-view components"]
```
Comment (Wumpf) [inline crates/re_space_view_spatial/src/visualizers/entity_iterator.rs:119]: it took me a bit to understand what you mean with a pov-component, haven't seen that term before. Is that a thing now or do we have a better name for it?

**Review by Wumpf — APPROVED** [review]
Looking great! Point visualizers will need way more love in the future, but your adjustments to status quo all make sense to me. Also, as briefly discussed on Slack, we'll eventually have to change things around to be more visual-range friendly overall (probably as part of the effort on better transform support there)

File: crates/re_space_view_spatial/src/visualizers/points3d.rs (hunk 2 above)
Comment (teh-cmc) [reply crates/re_space_view_spatial/src/visualizers/points3d.rs:345]: Yeah that's the remains of failed experiments, I'll revert that

**Pull request merged**